springwolf-plugins/springwolf-spring-cloud-rabbit-binder-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/CloudStreamRabbitBinderChannelsScanner.java

```diff
@@ -0,0 +1,198 @@
+import io.github.stavshamir.springwolf.asyncapi.types.channel.operation.message.PayloadReference;
+import io.github.stavshamir.springwolf.configuration.AsyncApiDocket;
+import io.github.stavshamir.springwolf.schemas.SchemasService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.amqp.core.Binding;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.binder.rabbit.properties.RabbitBinderConfigurationProperties;
+import org.springframework.cloud.stream.binder.rabbit.properties.RabbitExtendedBindingProperties;
+import org.springframework.cloud.stream.config.BindingServiceProperties;
+import org.springframework.context.EmbeddedValueResolverAware;
+import org.springframework.context.annotation.Bean;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringValueResolver;
+import sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+
+@Slf4j
+@Service
+@EnableConfigurationProperties({RabbitBinderConfigurationProperties.class, RabbitExtendedBindingProperties.class, BindingServiceProperties.class})
+public class CloudStreamRabbitBinderChannelsScanner implements ChannelsScanner, EmbeddedValueResolverAware {
+
+    private StringValueResolver resolver;
+    private final Map<String, Binding> bindingsMap;
+    private RabbitExtendedBindingProperties rabbitExtendedBindingProperties;
+    private BindingServiceProperties bindingServiceProperties;
+    private SchemasService schemasService;
+    private AsyncApiDocket docket;
+
+    public CloudStreamRabbitBinderChannelsScanner(AsyncApiDocket docket, RabbitExtendedBindingProperties rabbitExtendedBindingProperties, BindingServiceProperties bindingServiceProperties, SchemasService schemasService) {
+        this.docket = docket;
+        this.rabbitExtendedBindingProperties = rabbitExtendedBindingProperties;
+        this.bindingServiceProperties = bindingServiceProperties;
+        this.bindingsMap = makeBindingMap(rabbitExtendedBindingProperties, bindingServiceProperties);
+        this.schemasService = schemasService;
+    }
+
+    private Map<String, Binding> makeBindingMap(RabbitExtendedBindingProperties rabbitExtendedBindingProperties, BindingServiceProperties bindingServiceProperties) {
+        Map<String, StreamRabbitBindingInfo> bindingInfoMap = new HashMap<>();
```

Comment:
I think we should wrap the entire method (or the caller) with try/catch. Uncaught exceptions here will fail the entire application